gemini/sample-apps/fashion-trends-prediction/app/pages/0_ Trend Prediction.py

```diff
@@ -0,0 +1,405 @@
+                    key=f"btn_{i}",
+                    on_click=btn_pressed_callback,
+                    args=(i,),
+                )
+            chk_btn_status_and_assign_colour(buttons)
+        with col2:
+            try:
+                index = state["btn_prsd_status"].index(True)
+            except ValueError:
+                index = -1
+
+            if index != -1:
+                selected_button = index
+
+                outfit = str(state["items"][selected_button])
+
+                if outfit not in state:
+                    state[outfit] = {}
+
+                with st.spinner("Loading..."):
+                    try:
+                        print(state.keys())
+                        if state[outfit] == {}:
+                            _imgs = predict_image(
+                                instance_dict={
+                                    "prompt": generate(
+                                        image_prompt.format(outfit=outfit)
+                                    )
+                                },
+                                parameters={
+                                    "sampleCount": 1,
+                                    "sampleImageSize": 512,
+                                    "aspectRatio": "1:1",
+                                },
+                            )
+
+                            state[outfit]["imgs"] = [
+                                io.BytesIO(base64.b64decode(_img["bytesBase64Encoded"]))
+                                for _img in _imgs
+                            ]
+
+                        if len(state[outfit]["imgs"]) > 0:
+                            st.image(state[outfit]["imgs"][0], use_column_width=True)
+
+                            st.download_button(
+                                label="Download",
+                                key=f"_btn_download_{str(0) + ' ' + outfit}",
+                                data=state[outfit]["imgs"][0],
+                                file_name="image.png",
+                            )
+
+                        else:
+                            st.markdown(
+                                """
+                                    <style>
+                                    .box-container {
+                                        border: 1px solid gray;
+                                        padding: 120px;
+                                        border-radius: 5px;
+                                        background-color: #f5f5f5;
+                                    }
+                                    </style>
+                                    <div class="box-container">
+                                        Image generation failed due to responsible AI restrictions.
+                                    </div>
+                                    """,
+                                unsafe_allow_html=True,
+                            )
+
+                    except Exception as error:
```

Comment:
Try to reduce some of this nesting to make the code easier to read.

Reply:
These nestings are due to nested UI components.